src/component/isolation.rs

```diff
@@ -142,39 +144,24 @@ impl ProcessIsolationController for DockerArchiveController {
         let c_in = canonicalize(pipe.component_input_file())?;
         let c_out = canonicalize(pipe.component_output_file())?;
 
-        // We're calling docker run, mounting the input and output pipes, then running the loaded
-        // image with their paths as parameters
-        // TODO: Factor out the "run" logic to docker/mod.rs
-        let docker_subprocess = call_docker_async(&[
-            "run",
-            "-v",
-            &format!("{}:{}", c_in, c_in),
-            "-v",
-            &format!("{}:{}", c_out, c_out),
-            &self.docker_image_tag,
-            &c_in,
-            &c_out,
-        ])?;
-
-        Ok(Box::new(PipedProcessHandle {
-            subprocess: docker_subprocess,
-            pipe,
+        let container = V9Container::start(pipe, &self.docker_image_tag, &[&c_in, &c_out])?;
```

Comment:
Probably not the right place for this comment, but since you're making a call here it's kinda relevant- not sure how I feel about the name V9Container, since really it's a wrapper for doing docker stuff

Reply:
(We talked in person)